projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -77,13 +77,72 @@ def set_properties_factory_method(
 
         :param properties_type: str
                 string identifying the type of properties being set.
-                This enable users to configure what type of properties implemenation to use.
+                This enables users to configure what type of properties implementation to use.
         :param properties_factory: Callable[[], IPropertiesServiceClient]
                method or interface that returns correctly initialized implementation of IPropertiesServiceClient interface
         """
         pass
 
 
+class ISecretsServiceClient:
+    """
+    Client interface for reading and writing job secrets.
+    Different backend services can implement it so secrets can be stored in different secret storage providers
+    """
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def read_secrets(self, job_name: str, team_name: str) -> Dict:
+        """
+        Read secrets from the backend - returns a dictionary of all secrets for the given data job.
+        :param team_name: The name of the team the job belongs to
+        :param job_name: the name of the job
+        """
+        pass
+
+    @abstractmethod
+    def write_secrets(self, job_name: str, team_name: str, secrets: Dict) -> Dict:
+        """
+        Write secrets to the backend service. It overwrites all secrets completely.
+        :param team_name: The name of the team the job belongs to
+        :param job_name: the name of the job
+        :param secrets: dictionary with the new job's secrets
+        :return secrets: dictionary with the secrets handled
+        """
+        pass
+
+
+ISecretsFactory = Callable[[], ISecretsServiceClient]
+
+
+class ISecretsRegistry(ABC):
+    """
+    Secrets registry to enable register implementation Secrets service
```

Comment:
nit: that sounds awkward to me 

```suggestion
        Registry to enable the registration of Secrets service implementations.
```